src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RandomNumberGeneratorImplementation.cs

```diff
@@ -2,57 +2,82 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+
 namespace System.Security.Cryptography
 {
     internal sealed partial class RandomNumberGeneratorImplementation : RandomNumberGenerator
     {
-        public override void GetNonZeroBytes(byte[] data)
+        public override void GetBytes(byte[] data)
         {
-            if (data == null)
-                throw new ArgumentNullException(nameof(data));
+            if (data == null) throw new ArgumentNullException(nameof(data));
+
+            GetBytesInternal(data, 0, data.Length);
+        }
 
-            GetBytes(data);
+        public override void GetBytes(byte[] data, int offset, int count)
+        {
+            if (data == null) throw new ArgumentNullException(nameof(data));
+            if (offset < 0) throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (count < 0) throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (offset + count > data.Length) throw new ArgumentException(SR.Argument_InvalidOffLen);
```

Comment:
Not sure where you were implying the off-by-one (before or rewritten). The rewritten is equivalent to the before version except for subtracting `offset` on both sides of the > sign.

 If `count != 0` then `count` must be <= `data.Length - offset` (not <)

i.e. this randomizes the last byte and `1 <= 3 - 2`
````
            var rng = RandomNumberGenerator.Create();
            byte[] ran = new byte[3];
            rng.GetBytes(ran, 2, 1);
````

If `count==0` then there's this weirdness (allowed)
````
            rng.GetBytes(new byte[0], 0, 0);
            rng.GetBytes(new byte[1], 1, 0);
            rng.GetBytes(new byte[2], 2, 0);
````